### ISSUE TITLE
"preferences" window is not available

### ISSUE THREAD

**Issue by IIVQ** [autokey/autokey#883]
### AutoKey is a Xorg application and will not function in a Wayland session. Do you use Xorg (X11) or Wayland?

Xorg

### Has this issue already been reported?

- [X] I have searched through the existing issues.

### Is this a question rather than an issue?

- [X] This is not a question.

### What type of issue is this?

UI/Usability

### Choose one or more terms that describe this issue:

- [ ] autokey triggers
- [ ] autokey-gtk
- [x] autokey-qt
- [ ] beta
- [ ] bug
- [ ] critical
- [ ] development
- [ ] documentation
- [ ] enhancement
- [ ] installation/configuration
- [ ] phrase expansion
- [ ] scripting
- [ ] technical debt
- [ ] user interface

### Which Linux distribution did you use?

Operating System: Kubuntu 22.10
KDE Plasma Version: 5.25.5
KDE Frameworks Version: 5.98.0
Qt Version: 5.15.6
Kernel Version: 5.19.0-46-generic (64-bit)
Graphics Platform: X11
Processors: 8 × Intel® Core™ i7-6700 CPU @ 3.40GHz
Memory: 15,4 GiB of RAM
Graphics Processor: Mesa Intel® HD Graphics 530
Manufacturer: MSI
Product Name: MS-7982
System Version: 1.0

### Which AutoKey GUI did you use?

Qt

### Which AutoKey version did you use?

0.95.10

### How did you install AutoKey?

sudo apt-get install autokey-qt

### Can you briefly describe the issue?

Autokey automatically starts, I don't want this. Upon googling I find to disable it in the "preferences" which should be under edit.

The "Edit" menu has no preferences entry:
![image](https://github.com/autokey/autokey/assets/14251459/3036f538-c17e-4682-9865-7bdd08716ce2)

As an aside: I don't understand how autokey autostarts. It is not in ~/.config/autostart:

```
iivq@Betsy:~/.config/autostart$ ls -la
total 16
drwxrwxr-x  2 tijmen tijmen 4096 Apr  3  2021 .
drwxr-xr-x 57 tijmen tijmen 4096 Jul 17 18:49 ..
-rw-rw-r--  1 tijmen tijmen  309 Jul 17 09:48 dropbox.desktop
-rw-rw-r--  1 tijmen tijmen  192 Jun 30  2020 remmina-applet.desktop
```

### Can the issue be reproduced?

Always

### What are the steps to reproduce the issue?

1. Open Autokey (via the "A" in the system tray)
2. Go to menu -> Edit
3. No preferences item

### What should have happened?

I expected a preferences window, like the one shown in https://github.com/autokey/autokey/issues/149

### What actually happened?

There is no preferences menu

**Comment by josephj11** [comment]
Welcome to the AutoKey community, IIVQ!

Google isn't always your friend... :)

It's under Settings->Configure AutoKey->Automatically start AutoKey after login.

Note that some desktop environments, notably, KDE, will notice that AutoKey was running when they shutdown and will restore it as part of their session restore after they are restarted. There is usually a list of applications you can tell it to ignore when restoring a session or you can tell it not to restore at all or to restore from a static saved session.

On my old KDE system, the file is

$HOME/.config/autostart/autokey.desktop

I also have a custom startup script that doesn't involve AutoKey

$HOME/.config/autostart-scripts/mystart

Because AutoKey can be started by session restore and by the above .desktop file, it is sometimes necessary to disable one of the two methods so it doesn't get started twice and issue an error message.

This is really more of a support question than a valid issue. Please see the boiler plate below on the best ways to get AutoKey support.

----

BTW: Whenever you have a question (as opposed to an issue) it's better to post it on our [list](https://groups.google.com/forum/#!forum/autokey-users) or on [Gitter](https://gitter.im/autokey/autokey). The list is more user oriented whereas more technical questions will get a better answer on Gitter.

Both of these venues have a number of active users, so you are likely to get more, better, and faster answers on them than here on GitHub where only our very busy developers and a few users see the posts. We also prefer to have the developers developing rather than taking time to answer questions and manage issues that the user community could handle for them.

New users are encouraged to spend some time browsing through our [wiki](https://github.com/autokey/autokey/wiki/). There are a number of helpful articles there as well as many example AutoKey scripts.

----

I notice that you are not running 0.96.0, our current release. This release contains a lot of bug fixes and a few new features. We highly encourage users to upgrade as soon as possible. You can find it [here](https://github.com/autokey/autokey/releases/) and instructions are [here](https://github.com/autokey/autokey/wiki/Installing).

The installation is pretty easy to do and we also provide .deb packages for users of Debian and Debian-derived distributions (Ubuntu, Mint, ...).

**Comment by Elliria** [comment]
> Note that some desktop environments, notably, KDE, will notice that AutoKey was running when they shutdown and will restore it as part of their session restore after they are restarted. There is usually a list of applications you can tell it to ignore when restoring a session or you can tell it not to restore at all or to restore from a static saved session.

This is somebody's idea of "good" default behavior in Kubuntu. For my sanity's sake, one of the first things I do after the operating system is installed is to make it boot to an empty session.

For Kubuntu: **Application menu or Application Launcher or Application Dashboard** → **Settings** → **System Settings** → **Startup and Shutdown** → **Desktop Session** → Change the "**When logging in**" option to "**Start with an empty session**" or "**Restore manually saved session**".

**Comment by josephj11** [comment]
@Elliria I prefer the session restore. In fact, it doesn't do enough for me and I have a custom bash script that makes sure everything I want running gets started and is on the desired virtual desktop. It has a switch that disables it for when I'm doing repeated boots, etc.

That's what's really cool about Linux (and, especially, KDE). We can each have it our way and change it whenever we want to.

**Comment by Elliria** [comment]
Yeah, I figured I'd include "Restore manually saved session" in the steps for just such a scenario. For the time being, an empty session works for me, but if I eventually end up having stuff that needs to be run on boot or that's in specific places, then that option would also be useful for me.

**Comment by IIVQ** [comment]
@josephj11 (and others)
All the discussions of what KDE's good behaviour for saving Desktop states, autokey is not in ~/.config/autostart.

Thx for helping me. I also checked the "settings -> configure autokey" menu. Repeatedly. And wanted to upload images there is no "Automatically start Autokey" option. 

**But it is there** ... And somehow I **completely** missed it ... multiple times (because I checked this a few weeks ago as well).
I am on antibiotics, didn't realize I was this far off. 

Thx for helping me. I'll go sit in the corner of shame for a while now.

![image](https://github.com/autokey/autokey/assets/14251459/1828ed99-d453-49b5-8cf2-e47ace1116c4)

Then a very, very low-priority suggestion: make the "Automatically start AutoKey after Login"-button left-aligned, as the 5 other buttons are...

**Comment by josephj11** [comment]
Go take some prebiotics and probiotics and get your second brain back up to speed. LOL - Not medical advice.

Left-aligning that option makes sense to me if we're in that  code for something else. It needs its own enhancement issue if you want us to remember to do it.

For now, you can close this issue as completed.

**Comment by Elliria** [comment]
If you look at each of the sections in that screenshot, they all have a title except for the Auto-start section. The titles are all centered. The checkbox line in the Auto-start section is centered, so it's being treated as a title. This should be easily remedied by adding a title.

I poked around in the code and it looks like the [generalsettings.ui](https://github.com/autokey/autokey/blob/master/lib/autokey/qtui/resources/ui/generalsettings.ui) file is the culprit.

Each section in the screenshot above is a widget.If you do a search for **title** on that page, you'll see that each of the **groupbox** widgets have a title line with a title that corresponds to those you see in the screenshot above, but the [autostart_groupbox]((https://github.com/autokey/autokey/blob/master/lib/autokey/qtui/resources/ui/generalsettings.ui#L107)) widget doesn't have a proper title. It's using a title element as the text for the checkbox ([see line 114 through line 116 on that page](https://github.com/autokey/autokey/blob/master/lib/autokey/qtui/resources/ui/generalsettings.ui#L114-L116)), which is being detected by AutoKey as the title for that widget, so AutoKey is helping us out by centering it for us.

I'll bet that changing **title** to **text** on that line and inserting a proper **title** element at the beginning of the [autostart_groupbox]((https://github.com/autokey/autokey/blob/master/lib/autokey/qtui/resources/ui/generalsettings.ui#L107)) widget will take care of it.

I'm going to go try that in a VM and see. If not, then someone more experienced can take a look at these notes and will know how to fix it.

What should we use as the title for that section?
* Automatic Start
* Auto-start
* Start-up
* Startup
* Start
* etc...?

### Update
I've now tried and tried and am not able to edit the code in the [generalsettings.ui](https://github.com/autokey/autokey/blob/master/lib/autokey/qtui/resources/ui/generalsettings.ui) file. This will need to be a "Help wanted" issue where someone more knowledgeable than me takes up the torch and runs with it. Hopefully the information I've provided here will be of some use.

**Comment by josephj11** [comment]
I like Automatic Start, but I'd also like a blank title if that works to fix the justification.
